devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -78,6 +81,17 @@ protected boolean beforeExecute() throws MojoExecutionException {
     protected void doExecute() throws MojoExecutionException {
 
         try {
+            boolean isNativeProfileEnabled = isNativeProfileEnabled();
+            boolean clearPackageTypeSysProp = false;
+            // Essentially what this does is to enable the native package type even if a different package type is set
+            // in application properties. This is done to preserve what users expect to happen when
+            // they execute "mvn package -Dnative" even if quarkus.package.type has been set in application.properties
+            if (isNativeProfileEnabled
+                    && !System.getProperties().containsKey(PACKAGE_TYPE_PROP)) {
```

Comment:
Kind of a minor remark, the system property could be checked first, if it's not set there won't be a need to go through the profiles.

Reply:
Done